ui/src/SizesForm/components/SizesForm.js

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import Form from 'react-bootstrap/Form';
+import FormGroup from 'react-bootstrap/FormGroup';
+import Dropdown from 'react-bootstrap/Dropdown'
+import './SizesForm.css'
+import DropBtn from '../../CustomElements/DropBtn.js'
+import CustomPopup from '../../CustomElements/CustomPopup.js'
+
+class SizesForm extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            userCM: null,
+            sizeList: [],
+            selectedValue: '',
+            sizeDesc: '',
+            sizeSent: '',
+            sizeDefault: (
+                <p>
+                    Resources will be set based on profiles configured by administrator
+                </p>
+            ),
+        }
+
+    }
+
+    updateConfigmap() {
+        fetch('/services/jsp-api/api/user/configmap', {method:'GET'})
+            .then(response => {
+                if (response.ok) {
+                    return response.json();
+                } 
+                else {
+                    throw new Error('Failed to fetch user cm');
+                }
+            })
+            .then(data => {
+                console.log("Received size: ", data['last_selected_size'])
+                this.state.selectedValue = data['last_selected_size']
+                this.setState({selectedValue: this.state.selectedValue}) // Horrible... but it works
+                this.isSizeCorrect()
+            }) 
+    }
+
+    updateSizes() {
+        console.log("Called update Sizes")
+        fetch('/services/jsp-api/api/sizes', {method: 'GET'})
+            .then(response => {
+                if (response.ok) {
+                    return response.json();
+                } 
+                else {
+                    throw new Error('Unknown error.');
+                }
+            })
+            .then(data => {
+                if (data.length > 3) {
+                    this.updateSizes()
+                }
+                else {
+                    this.setState({sizeList: data}, this.updateConfigmap());
+                }
+            })
+    }
+
+    generateSizeDesc(event) {
+        var value = event.target.text
+        var json_data = {}
+        var result = ''
+        fetch('/services/jsp-api/api/size/'+ value, {method: 'GET'})
+        .then(response => {
+            if (response.ok) {
+                return response.json();
+            } 
+            else {
+                throw new Error('Unknown error.');
+            }
+        })
+        .then(data => {
+            json_data = data
+            result = <>
+                Size name: {json_data.name}<br/>
+                Limits:<br/>
+                &nbsp;&nbsp;&nbsp;CPU: {json_data.resources.limits.cpu} <br/>
```

Comment:
Let's keep it this way for now, but we should come up with a better way of formatting the popup content